### PR TITLE
adding mall

### PR DESCRIPTION
**Pull request by dahler**
adding mall as per excercise in class on Monday

### CODE REVIEW

File: 01-python/leisure/mall.py [hunk 1]
```diff
@@ -1,3 +1,54 @@
-def draw_mall():
-    print("mall not found")
+
+
+def draw_mall(height=2, width=3):
+    print("this is  the mall")
```
Comment (gallettilance) [inline 01-python/leisure/mall.py]: you should only print this if the input is valid

File: 01-python/leisure/mall.py [hunk 2]
```diff
@@ -1,3 +1,54 @@
-def draw_mall():
-    print("mall not found")
+
+
+def draw_mall(height=2, width=3):
```
Comment (gallettilance) [inline 01-python/leisure/mall.py]: please add a docstring to your function to explain what it does, what the input and output is etc.

File: 01-python/leisure/mall.py [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+    draw_mall_base(width)
+
+    if height > 0 and width > 0:
+        print()
+
     return
+
+def draw_mall_floor(width=3):
+    if width <= 0:
+       raise ValueError("Please input valid  width")
```
Comment (gallettilance) [inline 01-python/leisure/mall.py]: Note: if you push these into your parent function (such that they are not available outside the scope of `draw_mall()`, then you don't need to do error handling here. Up to you to decide!

**Comment by dahler** [comment]
> 
> 
> Looks good! Please squash your commits after addressing feedback

I am not sure how to squash commit

**Comment by gallettilance** [comment]
@dahler https://github.com/wprig/wprig/wiki/How-to-squash-commits

**Comment by dahler** [comment]
> 
> 
> @dahler https://github.com/wprig/wprig/wiki/How-to-squash-commits

Thank you, I managed to squash 3 commits to 1 commit. Thank you for the link

File: 01-python/leisure/mall.py [hunk 4]
```diff
@@ -1,3 +1,58 @@
...
+
+    for _ in range(height):
+        print()
+        draw_mall_floor(width)
+        print()
+        draw_mall_wall(width)
+    print()
+    draw_mall_base(width)
+
+    if height > 0 and width > 0:
```
Comment (gallettilance) [inline 01-python/leisure/mall.py]: this is always true right?

Comment (dahler) [reply 01-python/leisure/mall.py]: yes, because I have checked for error above

Comment (dahler) [reply 01-python/leisure/mall.py]: Do I need to recommit ?

Comment (gallettilance) [reply 01-python/leisure/mall.py]: only if you want this code to be merged

Comment (gallettilance) [reply 01-python/leisure/mall.py]: please use `git commit --amend` to avoid creating an extra commit

Comment (dahler) [reply 01-python/leisure/mall.py]: I did amend the commit. I am sorry that I did many times. It seems that my first 2 commit amend did not actually update the file.

**Pull request merged**